daml-lf/archive/BUILD.bazel

```diff
@@ -17,49 +17,79 @@ load(
 load("//bazel_tools:pom_file.bzl", "pom_file")
 load("@os_info//:os_info.bzl", "is_windows")
 load("//bazel_tools:javadoc_library.bzl", "javadoc_library")
+load("//bazel_tools:java.bzl", "da_java_proto_library")
+load("@com_github_johnynek_bazel_jar_jar//:jar_jar.bzl", "jar_jar")
 
 LF_MAJOR_VERSIONS = [
     "0",
     "1",
 ]
 
 filegroup(
-    name = "daml-lf-archive-srcs",
-    srcs = ["src/main/protobuf/com/digitalasset/daml_lf/daml_lf.proto"] +
-           ["src/main/protobuf/com/digitalasset/daml_lf/daml_lf_%s.proto" % v for v in LF_MAJOR_VERSIONS],
-    visibility = ["//visibility:public"],
-)
-
-exports_files(
-    ["com/digitalasset/daml_lf/daml_lf_%s.proto" % v for v in LF_MAJOR_VERSIONS],
+    name = "daml_lf_archive_proto_srcs",
+    srcs = glob(["src/main/protobuf/com/digitalasset/daml_lf/*.proto"]),
 )
 
 proto_library(
-    name = "daml_lf_proto",
-    srcs = [":daml-lf-archive-srcs"],
+    name = "daml_lf_archive_proto",
+    srcs = ["daml_lf_archive_proto_srcs"],
     strip_import_prefix = "src/main/protobuf/",
-    visibility = ["//:__subpackages__"],
+    visibility = ["//visibility:public"],
 )
 
 proto_gen(
-    name = "daml_lf_java_proto_srcs",
-    srcs = [":daml_lf_proto"],
+    name = "daml_lf_archive_java_proto_srcs",
+    srcs = ["daml_lf_archive_proto"],
     plugin_name = "java",
+    visibility = ["//visibility:public"],
 )
 
 java_library(
-    name = "daml_lf_java_proto",
-    srcs = [":daml_lf_java_proto_srcs"],
-    tags = ["maven_coordinates=com.digitalasset:daml-lf-archive:__VERSION__"],
+    name = "daml_lf_archive_java_proto",
+    srcs = ["daml_lf_archive_java_proto_srcs"],
+    tags = ["maven_coordinates=com.digitalasset:daml-lf-%s-archive-java-proto:__VERSION__"],
```

Comment:
What is the purpose of the `%s` here? Regardless of the details changing the maven coordinates will mean that we upload it to a different location so this should at the very least be documented in the release notes as users need to update this as well.

Reply:
My mistake. fixed